### ISSUE TITLE
Failed to create private endpoint to keyvault using azure_rm_privateendpoint

### ISSUE THREAD

**Issue by jinalklaulitz** [ansible/ansible_collections_azure#1234]
##### SUMMARY
Unable to create a private endpoint using azure_rm_privateendpoint for a keyvault.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure_rm_privateendpoint

##### ANSIBLE VERSION

```paste below
ansible [core 2.14.6]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  ansible collection location = /home/runner/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.16 (main, May 31 2023, 12:21:58) [GCC 8.5.0 20210514 (Red Hat 8.5.0-18)] (/usr/bin/python3.9)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
# /usr/share/ansible/collections/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 1.16.0 
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = /etc/ansible/ansible.cfg
```

##### OS / ENVIRONMENT

RHEL 9.2
Ansible Automation Platform 2.3-2
Ran on an image based on registry.redhat.io/ansible-automation-platform-23/ee-minimal-rhel8:latest

##### STEPS TO REPRODUCE

Run the following line:

```yaml
---
- name: Create private endpoint
  run_once: yes
  delegate_to: localhost
  block:
    - name: Create private endpoint
      azure.azcollection.azure_rm_privateendpoint:
        name: "{{ item.KVname }}-{{ ansibleRG }}-pe"
        resource_group: "{{ ansibleRG }}"
        subscription_id: "{{ SubscriptionID }}"
        state: present
        private_link_service_connections:
          - name: "{{ item.KVname }}-{{ ansibleRG }}-pe"
            private_link_service_id: "/subscriptions/{{ SubscriptionID }}/resourceGroups/{{ item.KVRG }}/providers/Microsoft.KeyVault/vaults/{{ item.KVname }}"
            group_ids: 
              - "vault"
        subnet:
          id: "/subscriptions/{{ SubscriptionID }}/resourceGroups/{{ ansibleNetworkRG }}/providers/Microsoft.Network/virtualNetworks/{{ ansibleNetworkVN }}/subnets/{{ ansibleNetworkPESubnet  }}"
      debugger: on_failed
```
Where: 
KVname = ansible-kv
KVRG = testingrg
ansibleRG = ansible
ansibleNetworkRG = network
ansibleNetworkVN = network-eastus
ansibleNetworkPESubnet = ansible-private-lin

All resources are in the same region, share the same virtual network and all exists. 

##### EXPECTED RESULTS

A private endpoint to a keyvault is created. 

##### ACTUAL RESULTS

```paste below
{
  "msg": "Error creating or updating private endpoint ansible-kv-ansible-pe - (InvalidResourceReference) Resource /subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/NETWORK/providers/Microsoft.Network/virtualNetworks/NETWORK-EASTUS referenced by resource /subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/ansible/providers/Microsoft.Network/privateEndpoints/ansible-kv-ansible-pe was not found. Please make sure that the referenced resource exists, and that both resources are in the same region.\nCode: InvalidResourceReference\nMessage: Resource /subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/NETWORK/providers/Microsoft.Network/virtualNetworks/NETWORK-EASTUS referenced by resource /subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/ansible/providers/Microsoft.Network/privateEndpoints/ansible-kv-ansible-pe was not found. Please make sure that the referenced resource exists, and that both resources are in the same region.\nException Details:\t(NotFound) Resource /subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/NETWORK/providers/Microsoft.Network/virtualNetworks/NETWORK-EASTUS not found.\n\tCode: NotFound\n\tMessage: Resource /subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/NETWORK/providers/Microsoft.Network/virtualNetworks/NETWORK-EASTUS not found.",
  "invocation": {
    "module_args": {
      "name": "ansible-kv-ansible-pe",
      "resource_group": "ansible",
      "subscription_id": "xxxx-xxxx-xxxx-xxxx",
      "private_link_service_connections": [
        {
          "name": "ansible-kv-ansible-pe",
          "private_link_service_id": "/subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/KVRG/providers/Microsoft.KeyVault/vaults/ansible-kv",
          "group_ids": [
            "vault"
          ]
        }
      ],
      "subnet": {
        "id": "/subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/network/providers/Microsoft.Network/virtualNetworks/network-eastus/subnets/ansible-private-lin"
      },
      "auth_source": "auto",
      "cloud_environment": "AzureCloud",
      "api_profile": "latest",
      "append_tags": true,
      "state": "present",
      "profile": null,
      "client_id": null,
      "secret": null,
      "tenant": null,
      "ad_user": null,
      "password": null,
      "cert_validation_mode": null,
      "adfs_authority_url": null,
      "log_mode": null,
      "log_path": null,
      "x509_certificate_path": null,
      "thumbprint": null,
      "tags": null,
      "location": null
    }
  },
  "_ansible_no_log": null,
  "changed": false,
  "_ansible_delegated_vars": {
    "ansible_host": "localhost",
    "ansible_port": null,
    "ansible_user": "root",
    "ansible_connection": "local"
  }
}
```

**Comment by Fred-sun** [comment]
@jinalklaulitz Judging from your error report, The reason for the error is not found '/ work/virtualNetworks/NETWORK - subscriptions/xxxx-xxxx-xxxx-xxxx/resourceGroups/NETWORK/providers/Microsoft.Net EASTUS 'resource does not exist. Could you please check if the resource exists? In particular, it is case sensitive. Thank you!
···e.g.
- name: Set Private Endpoint Name
  ansible.builtin.set_fact:
    rpfx: testkk0key4
    tenant_id: xxxxxxxxxxxxxxxxxxxxxxxx
    azure_client_id: xxxxxxxxxxxxxxxxxxxxxxxx
    azure_tenant: xxxxxxxxxxxxxxxxxxxxxxxx
    azure_secret: xxxxxxxxxxxxxxxxxxxxxxxx
  run_once: true

- name: Lookup service principal object id
  ansible.builtin.set_fact:
    object_id: "{{ lookup('azure_service_principal_attribute',
                   azure_client_id=azure_client_id,
                   azure_secret=azure_secret,
                   azure_tenant=tenant_id) }}"
  register: object_id_facts

- name: Create instance of Key Vault
  azure_rm_keyvault:
    resource_group: "{{ resource_group }}"
    vault_name: "vault{{ rpfx }}"
    enabled_for_deployment: true
    vault_tenant: "{{ tenant_id }}"
    sku:
      name: standard
      family: A
    access_policies:
      - tenant_id: "{{ tenant_id }}"
        object_id: "{{ object_id }}"
        keys:
          - get
          - list
          - update
          - create
          - import
          - delete
          - recover
          - backup
          - restore
        secrets:
          - get
          - list
          - set
          - delete
          - recover
          - backup
          - restore
          - purge
  register: vault_output

- name: Create virtual network
  azure_rm_virtualnetwork:
    name: vnet{{ rpfx }}
    address_prefixes_cidr:
      - 10.1.0.0/16
    dns_servers:
      - 127.0.0.1
      - 127.0.0.3
    resource_group: "{{ resource_group }}"

- name: Create network security group
  azure_rm_securitygroup:
    name: secgroup{{ rpfx }}
    resource_group: "{{ resource_group }}"

- name: Create the subnet
  azure_rm_subnet:
    name: subnet{{ rpfx }}
    virtual_network_name: vnet{{ rpfx }}
    resource_group: "{{ resource_group }}"
    address_prefix_cidr: "10.1.0.0/16"
    security_group: secgroup{{ rpfx }}
    private_endpoint_network_policies: Disabled
    private_link_service_network_policies: Disabled
    service_endpoints:
      - service: Microsoft.Sql
        locations:
          - eastus
          - westus
  register: subnet_output

- name: Create private endpoint
  azure_rm_privateendpoint:
    name: privateendpoint{{ rpfx }}
    resource_group: "{{ resource_group }}"
    private_link_service_connections:
      - name: privateEndpoints_test_name
        private_link_service_id: "{{ vault_output.id }}"
        group_ids:
          - vault
    subnet:
      id: "{{ subnet_output.state.id }}"
    tags:
      key1: value1
    cert_validation_mode: ignore
  register: output

- name: Get private endpoint info
  azure_rm_privateendpoint_info:
    name: privateendpoint{{ rpfx }}
    resource_group: "{{ resource_group }}"
  register: output

**Comment by Fred-sun** [comment]
@jinalklaulitz The azure. azcolleciton has released version v2.0.0. Please update and retest it! Thank you very much!
